### PR TITLE
Update docker configuration for AWS EC2 deployment

### PR DESCRIPTION
**Pull request by goneall**
Includes a README-PRODUCTION.md which describes the process for deployment.

This is currently running at http://52.32.53.255/

### CODE REVIEW

**Comment by goneall** [comment]
@kmehant @Ugtan @rnjudge Please review.  Since I'm not very expert in Docker configs, I'm very open to suggestions for improvement.

File: docker-compose.prod.yml [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+    expose:
+      - 8000
     volumes:
       - ./container_logs:/spdx/container_logs
+    env_file: spdx-prod.env
     environment:
       - PYTHONDONTWRITEBYTECODE=1
       - SPDX_REDIS_HOST=redis
+      - SPDXParser.LocalLicensesDir=/licenses/current
+      - SPDXParser.OnlyUseLocalLicenses=true
```
Comment (rtgdk) [inline docker-compose.prod.yml:29]: Where are these variables used?

**Comment by rtgdk** [comment]
@goneall Left two comments. Other than that, PR looks good to merge.

File: docker-compose.prod.yml (hunk 1 above)
Comment (goneall) [reply docker-compose.prod.yml:29]: I updated the tools to a new version which supports a cached version of the license to resolve the timeouts for license matching (see https://github.com/spdx/tools/pull/251)  These environment variables provide the location of the cached licenses.

File: dockerfile.dev [hunk 2]
```diff
@@ -13,6 +13,12 @@ FROM openjdk:8
 # Copy complete python container into java container
 COPY --from=python-container / /
 
+# Fetch the latest tagged licenses from Github for use in the Java license compares
+RUN apt-get update \
+    && apt-get -y install git
+RUN mkdir /licenses && cd /licenses && git clone https://github.com/spdx/license-list-data.git --depth=1 && cd license-list-data && git fetch --tags && git checkout $(git describe --tags `git rev-list --tags --max-count=1`)
+RUN mkdir /licenses/current && cp /licenses/license-list-data/website/*.jsonld /licenses/current && cp /licenses/license-list-data/website/licenses.json /licenses/current && cp /licenses/license-list-data/website/exceptions.json /licenses/current
+
```
Comment (kmehant) [inline dockerfile.dev:21]: We can add ` rm -rf /var/lib/apt/lists/*` for cleaning up.

File: dockerfile.prod [hunk 3]
```diff
@@ -33,6 +40,12 @@ FROM openjdk:8
 # Copy complete python container into java container
 COPY --from=production / /
 
+# Fetch the latest tagged licenses from Github for use in the Java license compares
+RUN apt-get update \
+    && apt-get -y install git
+RUN mkdir /licenses && cd /licenses && git clone https://github.com/spdx/license-list-data.git --depth=1 && cd license-list-data && git fetch --tags && git checkout $(git describe --tags `git rev-list --tags --max-count=1`)
+RUN mkdir /licenses/current && cp /licenses/license-list-data/website/*.jsonld /licenses/current && cp /licenses/license-list-data/website/licenses.json /licenses/current && cp /licenses/license-list-data/website/exceptions.json /licenses/current
+
```
Comment (kmehant) [inline dockerfile.prod:48]: We can add ` rm -rf /var/lib/apt/lists/*` for cleaning up.

File: nginx/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,4 @@
+FROM nginx:1.19.0-alpine
+
+RUN rm /etc/nginx/conf.d/default.conf
+COPY nginx.conf /etc/nginx/conf.d
```
[the remark this answers is not in the document]
Comment (goneall) [reply nginx/Dockerfile]: Added the newline

File: nginx/nginx.conf [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    charset utf-8;
+	client_max_body_size 75M;
+    location / {
+        proxy_pass http://spdx;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header Host $host;
+        proxy_redirect off;
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (goneall) [reply nginx/nginx.conf]: Added the newline

File: docker-compose.prod.yml [hunk 6]
```diff
@@ -1,18 +1,32 @@
-version: "3"
+version: "3.7"
 
 services:
+
+  nginx:
+    build: ./nginx
```
[the remark this answers is not in the document]
Comment (goneall) [reply docker-compose.prod.yml:7]: Added the container name

File: dockerfile.dev (hunk 2 above)
Comment (goneall) [reply dockerfile.dev:21]: Added the suggested command

**Comment by goneall** [comment]
Thanks @kmehant for the suggestions - I just updated the scripts.

**Comment by kmehant** [comment]
@goneall Looks good to me

**Pull request merged**